From c82aca6c3b4ab3aca97cba46e2fa0bc0369a37a0 Mon Sep 17 00:00:00 2001
From: Peter M <petermm@gmail.com>
Date: Thu, 15 Oct 2026 13:48:44 +0200
Subject: [PATCH] bsps/riscv: Add ESP32-C5 BSP (riscv/esp32c5db)

Add ESP32-C5 board support package (riscv/esp32c5db) with rv32imac
architecture and ilp32 ABI.

- Map unified 384 KiB HP SRAM (0x40800000 - 0x40860000)
- Set direct-boot signature header and flash window at 0x42000000
- Reserve 0x16000 (88 KiB) at the top of HP SRAM for ROM data, BSS, and stack
- Add ESP32-C5 interrupt matrix and INTPRI routing with 84 peripheral sources
- Adapt systimer, GPIO matrix (ngpios = 29), SPI, I2C, and watchdog registers
- Enable BSP_START_COPY_SECTIONS for flash data copying at boot

Signed-off-by: Peter M <petermm@gmail.com>
---
 bsps/riscv/esp32/bt/bt_c3.c                   |   4 +-
 bsps/riscv/esp32/i2c/i2c_c3.c                 |   5 +-
 .../riscv/esp32/include/c5/chip_definitions.h | 183 ++++++++++++++++++
 bsps/riscv/esp32/irq/irq_c3.c                 |  16 +-
 bsps/riscv/esp32/start/bspstart.c             |   4 +-
 spec/build/bsps/riscv/esp32/abi.yml           |   4 +
 spec/build/bsps/riscv/esp32/bspesp32c5db.yml  |  23 +++
 .../build/bsps/riscv/esp32/optchipvariant.yml |   2 +
 .../riscv/esp32/optdataflashmappedbase.yml    |   2 +
 .../bsps/riscv/esp32/optdramromreserve.yml    |   2 +
 spec/build/bsps/riscv/esp32/optiramdelta.yml  |   2 +
 .../bsps/riscv/esp32/optramcodeorigin.yml     |   2 +
 spec/build/bsps/riscv/esp32/optramorigin.yml  |   2 +
 spec/build/bsps/riscv/esp32/optramsize.yml    |   2 +
 spec/build/bsps/riscv/esp32/optvariant.yml    |   2 +
 .../bsps/riscv/optbspstartcopysections.yml    |   1 +
 .../bsps/riscv/optespressifdirectboot.yml     |   1 +
 17 files changed, 244 insertions(+), 13 deletions(-)
 create mode 100644 bsps/riscv/esp32/include/c5/chip_definitions.h
 create mode 100644 spec/build/bsps/riscv/esp32/bspesp32c5db.yml

diff --git a/bsps/riscv/esp32/bt/bt_c3.c b/bsps/riscv/esp32/bt/bt_c3.c
index 1271e55..b37373a 100644
--- a/bsps/riscv/esp32/bt/bt_c3.c
+++ b/bsps/riscv/esp32/bt/bt_c3.c
@@ -40,7 +40,7 @@
 
 #include <rtems/counter.h>
 
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
 
 void bsp_esp32_bt_get_state( bsp_esp32_bt_state *state )
 {
@@ -236,4 +236,4 @@ rtems_status_code bsp_esp32_bt_disable( void )
   return RTEMS_SUCCESSFUL;
 }
 
-#endif /* !defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) */
+#endif /* !defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) && !defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H) */
diff --git a/bsps/riscv/esp32/i2c/i2c_c3.c b/bsps/riscv/esp32/i2c/i2c_c3.c
index f9630e0..cda7503 100644
--- a/bsps/riscv/esp32/i2c/i2c_c3.c
+++ b/bsps/riscv/esp32/i2c/i2c_c3.c
@@ -162,7 +162,10 @@
 #define IO_MUX_MCU_SEL_MASK   ( 0x7U << IO_MUX_MCU_SEL_SHIFT )
 #define IO_MUX_FUNC_GPIO      1
 
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
+#define I2C_SIG_SCL 46
+#define I2C_SIG_SDA 47
+#elif defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
 #define I2C_SIG_SCL 45
 #define I2C_SIG_SDA 46
 #else
diff --git a/bsps/riscv/esp32/include/c5/chip_definitions.h b/bsps/riscv/esp32/include/c5/chip_definitions.h
new file mode 100644
index 0000000..e41ada3
--- /dev/null
+++ b/bsps/riscv/esp32/include/c5/chip_definitions.h
@@ -0,0 +1,183 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+/**
+ * @file
+ *
+ * @ingroup ESP32
+ *
+ * @brief Interrupt and address definitions specific to ESP32-C5.
+ */
+
+/*
+ * Copyright (C) 2026 Kinsey Moore <wkmoore@gmail.com>
+ * Copyright (C) 2026 Peter M. <petermm@gmail.com>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#ifndef LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H
+#define LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H
+
+#ifndef ASM
+
+#include <stdint.h>
+
+#endif /* ASM */
+
+/*
+ * 1-31 for the internal VIC with IRQ 0 being invalid since it corresponds to
+ * machine exception. This controller is fronted by an interrupt matrix
+ * supporting 84 peripheral interrupts (0-83). The count of 85 below includes
+ * 84 peripheral interrupts plus invalid interrupt 0.
+ */
+#define RISCV_MAXIMUM_EXTERNAL_INTERRUPTS 85
+#define BSP_INTERRUPT_VECTOR_COUNT        RISCV_MAXIMUM_EXTERNAL_INTERRUPTS
+
+#define BSP_INTERRUPT_CUSTOM_VALID_VECTOR
+
+#define BSP_GPIO_PIN_COUNT     29
+#define BSP_GPIO_PINS_PER_BANK 32
+
+/*
+ * The WiFi sources, which are peripheral interrupts 0 to 3 on this part.
+ */
+#define WIFI_MAC_INTR          0
+#define WIFI_MAC_NMI_INTR      1
+#define WIFI_PWR_INTR          2
+#define WIFI_BB_INTR           3
+
+/*
+ * Bluetooth and Low Power sources.
+ */
+#define BT_MAC_INTR            4
+#define BT_BB_INTR             5
+#define BT_BB_NMI_INTR         6
+#define LP_TIMER_INTR          7
+#define COEX_INTR              8
+#define BLE_TIMER_INTR         9
+#define BLE_SEC_INTR          10
+#define I2C_MASTER_INTR       11
+#define ZB_MAC_INTR           12
+#define PMU_INTR              13
+#define EFUSE_INTR            14
+#define LP_RTC_TIMER_INTR     15
+#define LP_UART_INTR          16
+#define LP_I2C_INTR           17
+#define LP_WDT_INTR           18
+#define LP_PERI_TIMEOUT_INTR  19
+#define LP_APM_M0_INTR        20
+#define LP_APM_M1_INTR        21
+#define HUK_INTR              22
+
+/*
+ * CPU software interrupts.
+ */
+#define SW_INTR_0             23
+#define SW_INTR_1             24
+#define SW_INTR_2             25
+#define SW_INTR_3             26
+
+#define ASSIST_DEBUG_INTR     27
+#define TRACE_INTR            28
+#define CACHE_INTR            29
+#define CPU_PERI_TIMEOUT_INTR 30
+#define GPIO_PROCPU_INTR      31
+#define GPIO_PROCPU_NMI_INTR  32
+#define PAU_INTR              33
+#define HP_PERI_TIMEOUT_INTR  34
+#define MODEM_PERI_TIMEOUT_INTR 35
+#define HP_APM_M0_INTR        36
+#define HP_APM_M1_INTR        37
+#define HP_APM_M2_INTR        38
+#define HP_APM_M3_INTR        39
+#define HP_APM_M4_INTR        40
+#define LP_APM0_INTR          41
+#define CPU_APM_M0_INTR       42
+#define CPU_APM_M1_INTR       43
+#define MSPI_INTR             44
+#define I2S_INTR              45
+#define UHCI0_INTR            46
+#define UART_INTR             47
+#define UART1_INTR            48
+#define LEDC_INTR             49
+#define TWAI0_INTR            50
+#define TWAI0_TIMER_INTR      51
+#define TWAI1_INTR            52
+#define TWAI1_TIMER_INTR      53
+#define USB_SERIAL_JTAG_INTR  54
+#define RMT_INTR              55
+#define I2C_EXT0_INTR         56
+#define TG_T0_INTR            57
+#define TG_WDT_INTR           58
+#define TG1_T0_INTR           59
+#define TG1_WDT_INTR          60
+#define SYSTIMER_TARGET0_INTR 61
+#define SYSTIMER_TARGET1_INTR 62
+#define SYSTIMER_TARGET2_INTR 63
+#define APB_ADC_INTR          64
+#define MCPWM0_INTR           65
+#define PCNT_INTR             66
+#define PARL_IO_TX_INTR       67
+#define PARL_IO_RX_INTR       68
+#define SLC0_INTR             69
+#define SLC1_INTR             70
+#define GDMA_CH0_INTR         71
+#define GDMA_CH1_INTR         72
+#define GDMA_CH2_INTR         73
+#define GDMA_OUT_CH0_INTR     74
+#define GDMA_OUT_CH1_INTR     75
+#define GDMA_OUT_CH2_INTR     76
+#define GPSPI2_INTR           77
+#define GPSPI2_INTR_2         77
+#define AES_INTR              78
+#define SHA_INTR              79
+#define RSA_INTR              80
+#define ECC_INTR              81
+#define ECDSA_INTR            82
+#define KM_INTR               83
+
+#define RISCV_INTERRUPT_VECTOR_SOFTWARE SW_INTR_0
+#define RISCV_INTERRUPT_VECTOR_TIMER    SYSTIMER_TARGET0_INTR
+
+#define GPIO_BASE            ( (uintptr_t) 0x60091000U )
+#define I2C_BASE             ( (uintptr_t) 0x60004000U )
+#define UART0_BASE           ( (uintptr_t) 0x60000000U )
+#define UART1_BASE           ( (uintptr_t) 0x60001000U )
+#define IO_MUX_BASE          ( (uintptr_t) 0x60090000U )
+#define SYSTIMER_BASE        ( (uintptr_t) 0x6000A000U )
+#define USB_SERIAL_JTAG_BASE ( (uintptr_t) 0x6000F000U )
+#define RTC_CNTL_BASE        ( (uintptr_t) 0x600B1000U )
+#define LP_AON_BASE          ( (uintptr_t) 0x600B1000U )
+#define LP_WDT_BASE          ( (uintptr_t) 0x600B1C00U )
+#define TIMG_BASE            ( (uintptr_t) 0x60008000U )
+#define INT_MATRIX_BASE      ( (uintptr_t) 0x60010000U )
+#define INTPRI_BASE          ( (uintptr_t) 0x600C5000U )
+#define SYSREG_BASE          ( (uintptr_t) 0x60095000U )
+#define PCR_BASE             ( (uintptr_t) 0x60096000U )
+#define SPI2_BASE            ( (uintptr_t) 0x60081000U )
+
+/* ESP32-C5 ROM function addresses */
+#define UART_TX_ONE_CHAR_ADDR ( (void *) (uintptr_t) 0x40000054 )
+#define UART_RX_ONE_CHAR_ADDR ( (void *) (uintptr_t) 0x40000060 )
+#define UART_TX_FLUSH_ADDR    ( (void *) (uintptr_t) 0x40000074 )
+
+#endif /* LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H */
diff --git a/bsps/riscv/esp32/irq/irq_c3.c b/bsps/riscv/esp32/irq/irq_c3.c
index 268cfc7..39b38b9 100644
--- a/bsps/riscv/esp32/irq/irq_c3.c
+++ b/bsps/riscv/esp32/irq/irq_c3.c
@@ -46,7 +46,7 @@ typedef struct peripheral_irq_map {
   uint8_t             cpu_int;
 } peripheral_irq_map_t;
 
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
 static peripheral_irq_map_t irq_mappings[] = {
   { .peripheral_int = WIFI_MAC_INTR, .cpu_int = 1 },
   { .peripheral_int = WIFI_PWR_INTR, .cpu_int = 1 },
@@ -176,7 +176,7 @@ static peripheral_irq_map_t irq_mappings[] = {
 
 #define MATRIX_REG( reg ) *( (volatile uint32_t *) ( INT_MATRIX_BASE + reg ) )
 
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
 /* ESP32-C6 Interrupt Matrix status registers */
 #define INTERRUPT_CORE0_INTR_STATUS_0_REG 0x134
 #define INTERRUPT_CORE0_INTR_STATUS_1_REG 0x138
@@ -290,7 +290,7 @@ static rtems_vector_number get_active_interrupt( uint8_t cpu_vector )
 {
   uint32_t s0 = MATRIX_REG( INTERRUPT_CORE0_INTR_STATUS_0_REG );
   uint32_t s1 = MATRIX_REG( INTERRUPT_CORE0_INTR_STATUS_1_REG );
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
   uint32_t s2 = MATRIX_REG( INTERRUPT_CORE0_INTR_STATUS_2_REG );
 #endif
 
@@ -305,7 +305,7 @@ static rtems_vector_number get_active_interrupt( uint8_t cpu_vector )
         if ( ( s1 & ( 1U << ( vec - 32 ) ) ) != 0 ) {
           return vec;
         }
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
       } else if ( vec < RISCV_MAXIMUM_EXTERNAL_INTERRUPTS ) {
         if ( ( s2 & ( 1U << ( vec - 64 ) ) ) != 0 ) {
           return vec;
@@ -443,7 +443,7 @@ rtems_status_code bsp_interrupt_is_pending(
     *pending = ( MATRIX_REG( INTERRUPT_CORE0_INTR_STATUS_0_REG ) & ( 1U << vector ) ) != 0;
   } else if ( vector < 64 ) {
     *pending = ( MATRIX_REG( INTERRUPT_CORE0_INTR_STATUS_1_REG ) & ( 1U << ( vector - 32 ) ) ) != 0;
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
   } else if ( vector < RISCV_MAXIMUM_EXTERNAL_INTERRUPTS ) {
     *pending = ( MATRIX_REG( INTERRUPT_CORE0_INTR_STATUS_2_REG ) & ( 1U << ( vector - 64 ) ) ) != 0;
 #endif
@@ -454,7 +454,7 @@ rtems_status_code bsp_interrupt_is_pending(
   return RTEMS_SUCCESSFUL;
 }
 
-#if !defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if !defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) && !defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
 #define SYSTEM_CPU_INTR_FROM_CPU_0_REG 0x28
 #define SYSTEM_CPU_INTR_FROM_CPU_1_REG 0x2C
 #define SYSTEM_CPU_INTR_FROM_CPU_2_REG 0x30
@@ -467,7 +467,7 @@ rtems_status_code bsp_interrupt_raise( rtems_vector_number vector )
 {
   bsp_interrupt_assert( bsp_interrupt_is_valid_vector( vector ) );
 
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
   switch ( vector ) {
     case SW_INTR_0:
       INTPRI_REG( 0x90 ) = 1;
@@ -520,7 +520,7 @@ rtems_status_code bsp_interrupt_clear( rtems_vector_number vector )
   esp32_cpu_int_clear( cpu_int );
 
   /* If this is a software interrupt, clear it. */
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
   switch ( vector ) {
     case SW_INTR_0:
       INTPRI_REG( 0x90 ) = 0;
diff --git a/bsps/riscv/esp32/start/bspstart.c b/bsps/riscv/esp32/start/bspstart.c
index 0d4d649..bb505d5 100644
--- a/bsps/riscv/esp32/start/bspstart.c
+++ b/bsps/riscv/esp32/start/bspstart.c
@@ -82,7 +82,7 @@ void bsp_start( void )
 {
   uint32_t timg_wdtconfig0;
 
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
   /* Unlock and disable LP WDT on C6 */
   WRITE_REG( LP_WDT_BASE, 0x18, 0x50d83aa1U );
   uint32_t lp_wdt = READ_REG( LP_WDT_BASE, 0x0 );
@@ -181,7 +181,7 @@ BSP_START_TEXT_SECTION static inline void copy_from_flash_offset(
 )
 {
   /* The RAM load sections are offset from 0x0, offset from mapped flash base */
-#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H)
+#if defined(LIBBSP_ESP32_C6_CHIP_DEFINITIONS_H) || defined(LIBBSP_ESP32_C5_CHIP_DEFINITIONS_H)
   uintptr_t flash_base = 0x42000000;
 #else
   uintptr_t flash_base = 0x3c000000;
diff --git a/spec/build/bsps/riscv/esp32/abi.yml b/spec/build/bsps/riscv/esp32/abi.yml
index 05ecc88..78471b2 100644
--- a/spec/build/bsps/riscv/esp32/abi.yml
+++ b/spec/build/bsps/riscv/esp32/abi.yml
@@ -12,6 +12,10 @@ default:
   value:
   - -march=rv32imc
   - -mabi=ilp32
+- enabled-by: riscv/esp32c5db
+  value:
+  - -march=rv32imac
+  - -mabi=ilp32
 - enabled-by: riscv/esp32c6db
   value:
   - -march=rv32imac
diff --git a/spec/build/bsps/riscv/esp32/bspesp32c5db.yml b/spec/build/bsps/riscv/esp32/bspesp32c5db.yml
new file mode 100644
index 0000000..89a450d
--- /dev/null
+++ b/spec/build/bsps/riscv/esp32/bspesp32c5db.yml
@@ -0,0 +1,23 @@
+SPDX-License-Identifier: CC-BY-SA-4.0 OR BSD-2-Clause
+arch: riscv
+bsp: esp32c5db
+build-type: bsp
+cflags: []
+copyrights:
+- Copyright (C) 2026 Kinsey Moore <wkmoore@gmail.com>
+- Copyright (C) 2026 Peter M. <petermm@gmail.com>
+cppflags: []
+enabled-by: true
+family: esp32
+includes: []
+install:
+- destination: ${BSP_INCLUDEDIR}/${ESPRESSIF_CHIP_VARIANT}
+  source:
+  - bsps/riscv/esp32/include/c5/chip_definitions.h
+links:
+- role: build-dependency
+  uid: ../../opto2
+- role: build-dependency
+  uid: grp
+source: []
+type: build
diff --git a/spec/build/bsps/riscv/esp32/optchipvariant.yml b/spec/build/bsps/riscv/esp32/optchipvariant.yml
index d893bcf..09723d2 100644
--- a/spec/build/bsps/riscv/esp32/optchipvariant.yml
+++ b/spec/build/bsps/riscv/esp32/optchipvariant.yml
@@ -9,6 +9,8 @@ copyrights:
 default:
 - enabled-by: riscv/esp32c3db
   value: 'c3'
+- enabled-by: riscv/esp32c5db
+  value: 'c5'
 - enabled-by: riscv/esp32c6db
   value: 'c6'
 - enabled-by: true
diff --git a/spec/build/bsps/riscv/esp32/optdataflashmappedbase.yml b/spec/build/bsps/riscv/esp32/optdataflashmappedbase.yml
index 7a80497..3849893 100644
--- a/spec/build/bsps/riscv/esp32/optdataflashmappedbase.yml
+++ b/spec/build/bsps/riscv/esp32/optdataflashmappedbase.yml
@@ -10,6 +10,8 @@ copyrights:
 default:
 - enabled-by: riscv/esp32c3db
   value: 0x3c000000
+- enabled-by: riscv/esp32c5db
+  value: 0x42000000
 - enabled-by: riscv/esp32c6db
   value: 0x42000000
 - enabled-by: true
diff --git a/spec/build/bsps/riscv/esp32/optdramromreserve.yml b/spec/build/bsps/riscv/esp32/optdramromreserve.yml
index 71fd8ca..d4a9b96 100644
--- a/spec/build/bsps/riscv/esp32/optdramromreserve.yml
+++ b/spec/build/bsps/riscv/esp32/optdramromreserve.yml
@@ -8,6 +8,8 @@ build-type: option
 copyrights:
 - Copyright (C) 2026 Samuel Price <thesamprice@gmail.com>
 default:
+- enabled-by: riscv/esp32c5db
+  value: 0x16000
 - enabled-by: riscv/esp32c6db
   value: 0x16000
 - enabled-by: true
diff --git a/spec/build/bsps/riscv/esp32/optiramdelta.yml b/spec/build/bsps/riscv/esp32/optiramdelta.yml
index c437957..4899a06 100644
--- a/spec/build/bsps/riscv/esp32/optiramdelta.yml
+++ b/spec/build/bsps/riscv/esp32/optiramdelta.yml
@@ -10,6 +10,8 @@ copyrights:
 default:
 - enabled-by: riscv/esp32c3db
   value: '0x3fc80000 - 0x40380000'
+- enabled-by: riscv/esp32c5db
+  value: '0'
 - enabled-by: riscv/esp32c6db
   value: '0'
 - enabled-by: true
diff --git a/spec/build/bsps/riscv/esp32/optramcodeorigin.yml b/spec/build/bsps/riscv/esp32/optramcodeorigin.yml
index 3ffb786..4d33442 100644
--- a/spec/build/bsps/riscv/esp32/optramcodeorigin.yml
+++ b/spec/build/bsps/riscv/esp32/optramcodeorigin.yml
@@ -10,6 +10,8 @@ copyrights:
 default:
 - enabled-by: riscv/esp32c3db
   value: 0x40380000
+- enabled-by: riscv/esp32c5db
+  value: 0x40800000
 - enabled-by: riscv/esp32c6db
   value: 0x40800000
 - enabled-by: true
diff --git a/spec/build/bsps/riscv/esp32/optramorigin.yml b/spec/build/bsps/riscv/esp32/optramorigin.yml
index 1fbd63e..7a8b76f 100644
--- a/spec/build/bsps/riscv/esp32/optramorigin.yml
+++ b/spec/build/bsps/riscv/esp32/optramorigin.yml
@@ -10,6 +10,8 @@ copyrights:
 default:
 - enabled-by: riscv/esp32c3db
   value: 0x3fc80000
+- enabled-by: riscv/esp32c5db
+  value: 0x40800000
 - enabled-by: riscv/esp32c6db
   value: 0x40800000
 - enabled-by: true
diff --git a/spec/build/bsps/riscv/esp32/optramsize.yml b/spec/build/bsps/riscv/esp32/optramsize.yml
index caa29cb..519ccc2 100644
--- a/spec/build/bsps/riscv/esp32/optramsize.yml
+++ b/spec/build/bsps/riscv/esp32/optramsize.yml
@@ -10,6 +10,8 @@ copyrights:
 default:
 - enabled-by: riscv/esp32c3db
   value: 0x50000
+- enabled-by: riscv/esp32c5db
+  value: 0x60000
 - enabled-by: riscv/esp32c6db
   value: 0x80000
 - enabled-by: true
diff --git a/spec/build/bsps/riscv/esp32/optvariant.yml b/spec/build/bsps/riscv/esp32/optvariant.yml
index ddb2544..4f7b672 100644
--- a/spec/build/bsps/riscv/esp32/optvariant.yml
+++ b/spec/build/bsps/riscv/esp32/optvariant.yml
@@ -10,6 +10,8 @@ copyrights:
 default:
 - enabled-by: riscv/esp32c3db
   value: 'c3/chip_definitions.h'
+- enabled-by: riscv/esp32c5db
+  value: 'c5/chip_definitions.h'
 - enabled-by: riscv/esp32c6db
   value: 'c6/chip_definitions.h'
 - enabled-by: true
diff --git a/spec/build/bsps/riscv/optbspstartcopysections.yml b/spec/build/bsps/riscv/optbspstartcopysections.yml
index debd719..c5b9cf8 100644
--- a/spec/build/bsps/riscv/optbspstartcopysections.yml
+++ b/spec/build/bsps/riscv/optbspstartcopysections.yml
@@ -9,6 +9,7 @@ copyrights:
 default:
 - enabled-by:
   - riscv/esp32c3db
+  - riscv/esp32c5db
   - riscv/esp32c6db
   value: true
 - enabled-by: true
diff --git a/spec/build/bsps/riscv/optespressifdirectboot.yml b/spec/build/bsps/riscv/optespressifdirectboot.yml
index ee32ac4..bc6181a 100644
--- a/spec/build/bsps/riscv/optespressifdirectboot.yml
+++ b/spec/build/bsps/riscv/optespressifdirectboot.yml
@@ -9,6 +9,7 @@ copyrights:
 default:
 - enabled-by:
   - riscv/esp32c3db
+  - riscv/esp32c5db
   - riscv/esp32c6db
   value: true
 - enabled-by: true
-- 
2.50.1 (Apple Git-155)

